### ISSUE TITLE
About to Launch App, met Unforeseen Problem

### ISSUE THREAD

**Issue by placetrend** [01org/appframework#864]
Hello, I'm 'Jake' Akhpan and I'm a developer with Placetrend. We are currently in Stealth Mode and are supposed to launch a travel pictures platform (built almost entirely using Intel App Framework 3.0) on Monday, May 12. Fortunately for us, one of our testers was kind enough to alert us to the fact that when the 'More Pictures' button at the bottom of a panel was tapped and new images loaded into the panel, the app doesn't automatically scroll to the top of the page. Our non-implementation of this feature was quite frankly an embarassing oversight on our part and we were extremely grateful to the tester in question for bringing it to our attention.

However we've gone through the IAF 3.0 API and there currently exists no option to add a scrollToTop functionality, and every single one of the hand-rolled and jQuery plugin-based scroll to top methods we tried simply refused to work.

The only method that currently works is when we use the: $(".scrollToThis").get(0).scrollIntoView(); Unfortunately this simply jumps the user to an element at the top of the page and does not provide smooth scrolling.

We would be grateful if we could be shown how to implement the scroll to top functionality in IAF 3.0 (we're already 2 weeks behind launch deadline, so we would be considerably grateful for any help on this particular issue). Thanks in advance.

'Jake' Akhpan,
Lead Developer,
Placetrend.

**Comment by imaffett** [comment]
Hi Jake,

This is an easy task.  You can bind the  panelbeforeload events and tell the panel to scroll to the top.  This will happen before the page transitions.  If you want smooth scrolling, use a third party plugin.

```

$("#pictures").on("panelbeforeload",function(){
   $("#pictures").get(0).scrollTop=0;
});
```

**Comment by placetrend** [comment]
Thanks for your prompt response. However it seems I was a bit vague in my explanation of the problem. The panel containing the pictures being viewed by the user (let's call it '#pictures') has already loaded and is displaying pictures.

Our problem is: when a user views the travel pictures on the '#pictures' panel, scrolls to the bottom of the [already loaded] pictures panel and taps a 'more pictures' button at the bottom of the panel, we run a function that loads pictures into this [already loaded] pictures panel ($.ajax to load the picture, and $().html to insert them into a div within the '#pictures' panel), however the app doesn't scroll the user back to the top of the '#pictures' panel after the pictures have been loaded into the panel, and none of the methods we've tried so far works (with the exception of '$("#pictures").get(0).scrollIntoView()' which merely jumps the user to the top of the view).

The '$(".scrollToThis").get(0).scrollIntoView()' which we currently use works adequately to return the user to the top of the page, the unsmooth jumping was our reason for looking for a possibly smoother, more native-like alternative. However if there isn't, we'll continue using the current method until we learn of a better alternative.

As a sidenote, I'd like to state that even taking the current issue into account, Intel App Framework 3.0 blows every other Javascript-based Mobile App Framework out of the water (we know because we've tried out practically every single framework on the market both jQuery and Angular based, they all promise a lot but always end up having multiple deal-breaking issues. In terms of ease of use, near-native handling, low learning curve, and multiplicity of useful features, mere words cannot describe the awesomeness of the Intel App Framework).

Once again, thanks for your prompt response. All the best.

Jake,
Placetrend.
